src/containerbuddy/main.go

```diff
@@ -48,10 +48,21 @@ func main() {
 		// Run our main application and capture its stdout/stderr.
 		// This will block until the main application exits and then os.Exit
 		// with the exit code of that application.
-		code, err := run(flag.Args())
+		config.Command = getCmd(flag.Args())
+		code, err := executeAndWait(config.Command)
 		if err != nil {
 			log.Println(err)
 		}
+		deregisterServices(config)
```

Comment:
Should this be called in the top of [`terminate`](https://github.com/joyent/containerbuddy/pull/22/files#diff-536bffba9bba22f8afea4ecb1036fe5cR30) in `signals.go` so that the requests to the service can be stopped before the service truly stops responding to them?

In this position, the shimmed app has been terminated before being deregistered, yes?